### PR TITLE
Fix long description overflow

### LINKED ISSUE #427: [BUG] Unable to see Title in a form with long description
### Issue Summary
I have created a form where there is a description which has a very long description. When I viewed that form, I am unable to see the title at the top. It is getting hidden at the top.

(Write your answer here.)

### Steps to Reproduce

1. Create a survey form with a question title and description
2. Give a very long description
3. Publish the survey
4. View the survey 
5. You won't be able to see the Question title
### Expected behavior
It should be scrollable and the title shouldn't be hidden at the top

#### Screenshots
https://app.formbricks.com/s/clivvfkvx0usomx0htzg5g7u7

#### Environment

- [x] Formbricks Cloud (app.formbricks.com)
- [ ] self-hosted Formbricks, version/commit: [please provide]

#### Desktop (please complete the following information):

- OS: Windows 11
- Browser: Chrome
- Version 114.0.5735.134

### PR DESCRIPTION
**Pull request by niteshseram**
## What does this PR do?
The preview in Link Survey and In-App Survey was not having any scroll behavior when the description or title are too long and it overflowed. This PR fixes those issues.

Fixes #427 

image
image
image

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

- [x] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [ ] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/SurveyMenuBar.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ export default function SurveyMenuBar({
           className="w-72 border-white hover:border-slate-200 "
         />
       </div>
-      {localSurvey?.responseRate && (
+      {!!localSurvey?.responseRate&& (
```
Comment (jobenjada) [inline apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/SurveyMenuBar.tsx:132]: Have you tested this?

I'm sceptical this "&&" operator works without the space.

And why did you invest this?

File: apps/web/app/s/[surveyId]/LinkSurvey.tsx [hunk 2]
```diff
@@ -50,7 +50,10 @@ export default function LinkSurvey({ survey }: LinkSurveyProps) {
           loadingElement && "animate-pulse opacity-60",
           "flex h-full flex-1 items-center overflow-y-auto bg-white"
         )}>
-        <ContentWrapper className="w-full md:max-w-lg">
+        <ContentWrapper className={cn(
+          isPreview && "mt-[44px]",
```
Comment (jobenjada) [inline apps/web/app/s/[surveyId]/LinkSurvey.tsx:54]: Please explain what you're solving here and why you chose this approach.

**Review by jobenjada — COMMENTED** [review]
Hey @niteshseram 

thanks a lot for your contribution :)

I left two comments in the code.

Regarding your overall solution: 

We had the same problem when a Select question had too many options to choose from. We decided to add the scroll to the options instead of the overall widget.

Why did you decide to add the scroll to the complete widget instead of the description field? :)

Thanks! 🙌

File: apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/SurveyMenuBar.tsx (hunk 1 above)
Comment (niteshseram) [reply apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/SurveyMenuBar.tsx:132]: @jobenjada This is a file That I didn't touch. Not sure, how this got pushed. Will invert this

File: apps/web/app/s/[surveyId]/LinkSurvey.tsx (hunk 2 above)
Comment (niteshseram) [reply apps/web/app/s/[surveyId]/LinkSurvey.tsx:54]: @jobenjada In Preview mode, there is a preview strip at the top and due to this some portion of the content get hidden below this preview strip
image

**Comment by niteshseram** [comment]
> Hey @niteshseram
> 
> thanks a lot for your contribution :)
> 
> I left two comments in the code.
> 
> Regarding your overall solution:
> 
> We had the same problem when a Select question had too many options to choose from. We decided to add the scroll to the options instead of the overall widget.
> 
> Why did you decide to add the scroll to the complete widget instead of the description field? :)
> 
> Thanks! 🙌

We can use that approach as well. But don't you think there will be too many scroll bars on one screen? Question and description doesn't limit, so there will be scroll for both again and there will be scroll for many options again.

**Comment by jobenjada** [comment]
I think you're right! 

I've played around with it and it makes more sense to have a global scroll vs. one for the single fields - definitely better UX, especially on smartphones.

1. Can you update your PR so that the scrolling for MultiSelect and SingleSelect are removed in both link surveys and the JS widget?
2. You did not yet add the scrolling to the JS widget
grafik

For the understanding (its a bit messy but we have a plan to unify the form components, just didnt get to it yet):

The components used in the JS widget are different from the components used in the Preview and the Link Survey. This is why you have to add the scrolling to the JS widget.

Please note: The JS widget is PREACT and does not take tailwind classes well. I already designed the scroll bar for the scrolling in the Select questions so adding `fb-overflow-y-auto` should already do the trick ;)

Thanks!!

**Comment by jobenjada** [comment]
Hey Nitesh,

here are the results of my testing:

1. Link Survey not centered anymore:
grafik

When you make your changes, please also test how the surveys behave when they are embedded by our iframe:
grafik

2. There is no margin-top in the JS widget:
grafik

Other than that, it looks great :)

**Comment by niteshseram** [comment]
Hi @jobenjada, Fixed the above-mentioned issues. Thanks

**Comment by jobenjada** [comment]
@mattinannt ready to merge 🚀

**Review by mattinannt — APPROVED** [review]
@niteshseram thanks a lot for this fix 🎉💪 Let's merge it into main 😊

**Pull request merged**